src/GafferAppleseed/IECoreAppleseedPreview/Renderer.cpp

```diff
@@ -1693,6 +1713,201 @@ class AppleseedDeltaLight : public AppleseedLight
 		asf::Transformd m_transform;
 };
 
+/// Appleseed area light handle.
+class AppleseedAreaLight : public AppleseedLight
+{
+	public :
+
+		AppleseedAreaLight( asr::Project &project, const string &name, const IECoreScenePreview::Renderer::AttributesInterface *attributes, IECoreScenePreview::Renderer::RenderType renderType )
+			:	AppleseedLight( project, name, attributes, renderType == IECoreScenePreview::Renderer::Interactive ) , m_transform( asf::Transformd::identity() ), m_renderType( renderType )
+		{
+			init();
+			AppleseedAreaLight::attributes( attributes );
+		}
+
+		virtual ~AppleseedAreaLight()
+		{
+			if( isInteractiveRender() )
+			{
+				removeAreaLightEntities();
+			}
+			else
+			{
+				// Create the material assignments.
```

Comment:
> If Appleseed is already capable of representing arbitrary mesh lights it might be really cool to expose that in the same way.

We have a simpler way I think. There's nothing special about a mesh light in appleseed.
Just assigning a shader with an emission closure will turn any object into a mesh light.
(For closed objects it's important to disable double sided material assignments).

![gaffer_std_surface](https://user-images.githubusercontent.com/409013/28975052-eb168aba-7938-11e7-80a3-9b7017baf4a3.png)